presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -143,6 +144,9 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
         if (!table.isPresent()) {
             return null;
         }
+        if (handle.getTableType() != DATA) {
+            throw new PrestoException(NOT_SUPPORTED, "Table type not yet supported: " + handle.getTableType());
```

Comment:
It may be good to provide the full table name instead of just the table type in the exception error message similar to `UnknownTableTypeException`.